nautobot/core/models/utils.py

```diff
@@ -260,3 +260,20 @@ def construct_natural_slug(values, pk=None):
     # Replace any emojis with their string name, and then slugify that.
     values = (slugify(emoji.replace_emoji(value, unicodedata.name)) for value in values)
     return constants.NATURAL_SLUG_SEPARATOR.join(values)
+
+
+def compress_range(iterable):
+    """
+    Generates compressed range from an un-sorted expanded range.
+    For example:
+        [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 100, 101, 102, 103, 104, 105, 1000, 1100, 1101, 1102, 1103, 1104, 1105, 1106]
+        =>
+        iter1: (1, 10)
+        iter2: (100, 105)
+        iter3: (1000, 1000)
+        iter4: (1100, 1106)
+    """
+    iterable = sorted(set(iterable))
+    for _, grp in groupby(enumerate(iterable), lambda t: t[1] - t[0]):
+        grp = list(grp)
```

Comment:
Why's this conversion necessary?

Reply:
_The returned group is itself an iterator that shares the underlying iterable with [groupby()](https://docs.python.org/3/library/itertools.html#itertools.groupby). Because the source is shared, when the [groupby()](https://docs.python.org/3/library/itertools.html#itertools.groupby) object is advanced, the previous group is no longer visible. So, if that data is needed later, it should be stored as a list:_

source: https://docs.python.org/3/library/itertools.html#itertools.groupby